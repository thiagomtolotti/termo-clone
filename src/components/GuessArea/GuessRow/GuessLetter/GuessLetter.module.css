.guess {
  width: 72px;
  aspect-ratio: 1;

  /* background: red; */
  color: white;
  font-weight: 600;
}

.guess.active {
  border-bottom-width: 11px !important;
}

.guess.correct {
  background-color: var(--green) !important;
}

.guess.misplaced {
  background-color: var(--yellow) !important;
}
