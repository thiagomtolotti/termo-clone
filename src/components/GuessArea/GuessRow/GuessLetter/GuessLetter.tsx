import clsx from "clsx";

interface GuessLetterProps {
  children: React.ReactNode;
  active?: boolean;
  isCorrectOrMisplaced?: "correct" | "misplaced";
  onClick(): void;
  isRowActive: boolean;
}

export const GuessLetter = ({
  active = false,
  children,
  isCorrectOrMisplaced,
  onClick,
  isRowActive,
}: GuessLetterProps) => {
  return (
    <div
      role="guess"
      className={clsx(
        "w-16 flex items-center justify-center font-bold aspect-square rounded-md text-5xl",
        isRowActive &&
          "bg-transparent border-4 border-brown-300 cursor-pointer",
        !isRowActive && Boolean(children) ? "bg-brown-100" : "bg-brown-400",
        active && "!border-b-[10px]",
        isCorrectOrMisplaced === "correct" && "bg-green",
        isCorrectOrMisplaced === "misplaced" && "bg-yellow"
      )}
      onClick={onClick}
    >
      {children}
    </div>
  );
};
