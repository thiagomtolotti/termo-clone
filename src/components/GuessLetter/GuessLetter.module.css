.guess {
  width: 40px;
  aspect-ratio: 1;

  background: red;
  color: white;
}
