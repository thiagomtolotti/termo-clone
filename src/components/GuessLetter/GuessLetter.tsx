"use client";

import { useState } from "react";
import styles from "./GuessLetter.module.css";

interface GuessLetterProps {
  active?: boolean;
}

const letterRegex = /^[a-zA-Z]$/;

export const GuessLetter = ({ active = false }: GuessLetterProps) => {
  const [value, setValue] = useState("");

  document.addEventListener("keydown", (ev) => {
    if (!active) return;

    const keyPressed = ev.key;

    if (keyPressed === "Backspace") {
      setValue("");
    }

    if (!letterRegex.test(keyPressed)) return;

    setValue(keyPressed.toUpperCase());
  });

  return (
    <div role="guess" className={styles.guess}>
      {value}
    </div>
  );
};
