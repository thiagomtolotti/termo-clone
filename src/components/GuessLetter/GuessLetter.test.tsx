import { fireEvent, render, screen } from "@testing-library/react";
import { GuessLetter } from "./GuessLetter";

// describe("GuessLetter component", () => {
//   const keyDownEvent = {
//     key: "T",
//     code: "T",
//   };
//   let guess: HTMLElement;

//   it("Should render correctly", () => {
//     const { container } = render(<GuessLetter />);

//     expect(container).toMatchSnapshot();
//   });

//   describe("if is active", () => {
//     beforeEach(() => {
//       render(<GuessLetter active />);
//       guess = screen.getByRole("guess");
//     });

//     it("If receives an input it should have it's inner value set to the input value", () => {
//       fireEvent.keyDown(document.body, keyDownEvent);

//       expect(guess.innerHTML).toBe(keyDownEvent.key);
//     });

//     it("If receives a new value it should substitute it's inner value", () => {
//       fireEvent.keyDown(document.body, keyDownEvent);

//       expect(guess.innerHTML).toBe(keyDownEvent.key);

//       const newKeyDownEvent = {
//         key: "Z",
//         code: "Z",
//       };
//       fireEvent.keyDown(document.body, newKeyDownEvent);

//       expect(guess.innerHTML).toBe(newKeyDownEvent.key);
//     });

//     it("Should convert the value of the key pressed to uppercase", () => {
//       const smallCaseKeyDownEvt = {
//         key: "t",
//         code: "t",
//       };

//       fireEvent.keyDown(document.body, smallCaseKeyDownEvt);

//       expect(guess.innerHTML).toBe(smallCaseKeyDownEvt.key.toUpperCase());
//     });

//     it("Should ignore inputs that are not letters", () => {
//       fireEvent.keyDown(document.body, keyDownEvent);

//       const AltKeyDownEvt = {
//         key: "Alt",
//         code: "Alt",
//       };

//       fireEvent.keyDown(document.body, AltKeyDownEvt);

//       expect(guess.innerHTML).toBe(keyDownEvent.code);
//     });

//     it("Should be empty when backspace is pressed", () => {
//       const backspaceKeyDownEvt = {
//         key: "Backspace",
//         code: "Backspace",
//       };

//       fireEvent.keyDown(document.body, keyDownEvent);
//       fireEvent.keyDown(document.body, backspaceKeyDownEvt);

//       expect(guess.innerHTML).toBe("");
//     });
//   });

//   it("If is not active and receives an input it should do nothing", () => {
//     render(<GuessLetter />);

//     const guess = screen.getByRole("guess");

//     fireEvent.keyDown(document.body, keyDownEvent);

//     expect(guess.innerHTML).toBe("");
//   });

//   describe("When active is toggled", () => {
//     it("Should disable the keyDown events when it isn't active", () => {
//       const { rerender } = render(<GuessLetter active />);

//       const guess = screen.getByRole("guess");
//       fireEvent.keyDown(document.body, keyDownEvent);

//       expect(guess.innerHTML).toBe(keyDownEvent.key);

//       rerender(<GuessLetter />);

//       const newKeyDownEvent = {
//         key: "Z",
//         code: "Z",
//       };

//       fireEvent.keyDown(document.body, newKeyDownEvent);

//       expect(guess.innerHTML).toBe(keyDownEvent.key);
//     });

//     it("Should enable the keyDown events when it is active", () => {
//       const { rerender } = render(<GuessLetter />);

//       const guess = screen.getByRole("guess");

//       fireEvent.keyDown(document.body, keyDownEvent);

//       expect(guess.innerHTML).toBe("");

//       rerender(<GuessLetter active />);

//       const newKeyDownEvent = {
//         key: "Z",
//         code: "Z",
//       };

//       fireEvent.keyDown(document.body, newKeyDownEvent);

//       expect(guess.innerHTML).toBe(newKeyDownEvent.key);
//     });
//   });
// });
