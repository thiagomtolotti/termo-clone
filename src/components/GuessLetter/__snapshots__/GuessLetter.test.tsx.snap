// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessLetter component Should render correctly 1`] = `
<div>
  <div
    class="guess"
    role="guess"
  />
</div>
`;
